test/controllers/HasSupplementaryUnitControllerSpec.scala

```diff
@@ -45,140 +49,332 @@ class HasSupplementaryUnitControllerSpec extends SpecBase with MockitoSugar {
+
+        running(application) {
+          val request = FakeRequest(GET, hasSupplementaryUnitRoute)
+
+          val result = route(application, request).value
+
+          status(result) mustEqual SEE_OTHER
+          redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+        }
+      }
+
+      "must redirect to Journey Recovery for a POST if no existing data is found" in {
+
+        val application = applicationBuilder(userAnswers = None).build()
+
+        running(application) {
+          val request =
+            FakeRequest(POST, hasSupplementaryUnitRoute)
+              .withFormUrlEncodedBody(("value", "true"))
+
+          val result = route(application, request).value
+
+          status(result) mustEqual SEE_OTHER
+          redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
+        }
       }
     }
+    ".update journey" - {
+      "must return OK and the correct view for a GET" in {
+
+        val mockGoodsRecordConnector = mock[GoodsRecordConnector]
+        when(mockGoodsRecordConnector.getRecord(any(), any())(any())) thenReturn Future
+          .successful(record)
+
+        when(mockGoodsRecordConnector.updateSupplementaryUnitForGoodsRecord(any(), any(), any())(any()))
+          .thenReturn(Future.successful(Done))
+
+        val application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
+          .overrides(bind[GoodsRecordConnector].toInstance(mockGoodsRecordConnector))
+          .build()
 
-    "must redirect to the next page when valid data is submitted" in {
+        running(application) {
+          val request = FakeRequest(GET, hasSupplementaryUnitUpdateRoute)
 
-      val mockSessionRepository = mock[SessionRepository]
+          val result = route(application, request).value
 
-      when(mockSessionRepository.set(any())) thenReturn Future.successful(true)
+          val view = application.injector.instanceOf[HasSupplementaryUnitView]
 
-      val application =
-        applicationBuilder(userAnswers = Some(emptyUserAnswers))
-          .overrides(
-            bind[Navigator].toInstance(new FakeNavigator(onwardRoute)),
-            bind[SessionRepository].toInstance(mockSessionRepository)
-          )
+          status(result) mustEqual OK
+          contentAsString(result) mustEqual view(form.fill(true), NormalMode, recordId, onSubmitUpdateAction)(
+            request,
+            messages(application)
+          ).toString
+        }
+      }
+
+      "must return OK and the correct view for a GET - should set to false when supplementary value is 0" in {
+
+        val mockGoodsRecordConnector = mock[GoodsRecordConnector]
+        when(mockGoodsRecordConnector.getRecord(any(), any())(any())) thenReturn Future
+          .successful(recordWithoutSuppUnit)
+
+        when(mockGoodsRecordConnector.updateSupplementaryUnitForGoodsRecord(any(), any(), any())(any()))
```

Comment:
This test doesn't need this mock right? this page shouldn't be updating goods records?

Reply:
you are a great spotter Chris! :)